feat: update MDM migration guide with new UX (#22128)
fleetdm/fleet: articles/mdm-migration.md

This guide provides instructions for migrating devices from your current MDM solution to Fleet.

> Migrating from an existing MDM solution to Fleet allows IT admins to consolidate device management and take advantage of Fleet's advanced monitoring and compliance features. This guide will help you smoothly transition your macOS devices to Fleet, whether they are manually enrolled or automatically enrolled (ADE).

Reviewer: ```suggestion
Migrating from an existing MDM solution to Fleet allows IT admins to consolidate device management and take advantage of Fleet's advanced monitoring and compliance features. This guide will help you smoothly transition your macOS devices to Fleet, whether they are manually enrolled or automatically enrolled (ADE).
```

Do we need these sentences? Less is more. We already have a sentence up here to describe the guide: https://github.com/fleetdm/fleet/pull/22128/files#diff-d2bf4696ec283bc0b492fd0ef70ea66c8e73a8192b2617a93c6999979e3f0c97R3

Also, one doesn't need to turn on MDM to use Fleet's monitoring and compliance features. Fleet offers MDM features is more than MDM.

Reviewer: I removed this sentence.

However, Activation Lock bypass codes can only be retrieved from the Mac up to 30 days after the device is enrolled. This means that when migrating from your old MDM solution, it’s likely that you’ll be unable to retrieve the Activation Lock bypass code.

### How to turn on disk encryption

1. Select the Fleet icon in your menu bar and select **My device**.

![Fleet icon in menu bar](https://raw.githubusercontent.com/fleetdm/fleet/main/website/assets/images/articles/fleet-desktop-says-hello-world-cover-1600x900@2x.jpg)

2. On your **My device** page, follow the disk encryption instructions in the yellow banner.
  - If you don’t see the yellow banner, select the purple **Refetch** button at the top of the page.
  - If you still don't see the yellow banner after a couple minutes or if the **My device** page presents you with an error, please contact your IT administrator.

<img width="1399" alt="My device page - turn on disk encryption" src="https://user-images.githubusercontent.com/5359586/229950451-cfcd2314-a993-48db-aecf-11aac576d297.png">

## Migrate manually enrolled hosts

> - **Manual Enrollment**: Devices that were manually enrolled in the previous MDM and need to be individually migrated to Fleet.

Reviewer: ```suggestion
- **Manual enrollment**: Devices that were manually enrolled in the previous MDM and need to be individually migrated to Fleet.
```

Sentence case: https://fleetdm.com/handbook/company/why-this-way#why-does-fleet-use-sentence-case

> - **Automatic Enrollment**: Devices enrolled via Apple Business Manager’s automated device enrollment, which can be migrated more seamlessly.

Reviewer: ```suggestion
- **Automatic enrollment**: Devices enrolled via Apple Business Manager’s automated device enrollment, which can be migrated more seamlessly.
```

This is because if the Activation Lock is enabled, you will need the Activation Lock bypass code to successfully wipe and reuse the Mac.

However, Activation Lock bypass codes can only be retrieved from the Mac up to 30 days after the device is enrolled. This means that when migrating from your old MDM solution, it’s likely that you’ll be unable to retrieve the Activation Lock bypass code.

### How to turn on disk encryption

1. Select the Fleet icon in your menu bar and select **My device**.

![Fleet icon in menu bar](https://raw.githubusercontent.com/fleetdm/fleet/main/website/assets/images/articles/fleet-desktop-says-hello-world-cover-1600x900@2x.jpg)

2. On your **My device** page, follow the disk encryption instructions in the yellow banner.
  - If you don’t see the yellow banner, select the purple **Refetch** button at the top of the page.
  - If you still don't see the yellow banner after a couple minutes or if the **My device** page presents you with an error, please contact your IT administrator.

> <img width="1399" alt="My device page - turn on disk encryption" src="https://user-images.githubusercontent.com/5359586/229950451-cfcd2314-a993-48db-aecf-11aac576d297.png">

Reviewer: Why move these to the top? I think as a reader of the article I want to learn how migration works first. Then I want to learn how migration affects FileVault and other items.

cc @spokanemac

Reviewer 2: Only some scrolls/reads to the bottom, and if you don't address FileVault and Activation lock ahead of the move, you'll have a tougher row to hoe. Happy to move it back down @noahtalerman.

Reviewer: Makes sense. I think up to you @spokanemac

```

2. Fill in the above keys under the `mdm.macos_migration` key.

To learn about each option, in the Fleet UI, select the avatar on the right side of the top navigation, select **Settings > Integrations > Mobile device management (MDM)**, and scroll down to the **End user migration workflow** section.

3. During the end user migration workflow, the window will show the Fleet logo on top of a dark and light background (appearance configured by end user).

If you would like to add your organization's logo, you can optionally set a logo for each background:

```yaml
apiVersion: v1
kind: config
spec:
  org_info:
    org_logo_url: https://fleetdm.com/images/press-kit/fleet-blue-logo.png
    org_logo_url_light_background: https://fleetdm.com/images/press-kit/fleet-white-logo.png
  ...
```

Add URLs to your logos that are visible on a dark background and light background in the `org_logo_url` and `org_logo_url_light_background` keys respectively. If you only set a logo for one, the Fleet logo will be used for the other.

1. Run the fleetctl `apply -f fleet-config.yml` command to add your configuration.

2. Confirm that your configuration was saved by running `fleetctl get config`.

> 3. Send [these guided instructions](#how-to-turn-on-mdm-end-user-manual-migration) to your end users to complete the final few steps via Fleet Desktop.

Reviewer: These steps look a little outdated (they use the old `fleetctl apply` YAML format).

I think let's update these to point to the best practice YAML format in the [YAML files (GitOps) reference docs](https://fleetdm.com/docs/configuration/yaml-files).

##### How to turn on MDM (end user)

1. Select the Fleet icon in your menu bar and select **Migrate to Fleet**.

2. Select **Start** in the **Migrate to Fleet** popup.

2. On your **My device** page, select the **Turn on MDM** button in the yellow banner and follow the instructions.
    * If you don’t see the yellow banner or the **Turn on MDM** button, select the purple **Refetch** button at the top of the page.
    * If you still don't see the **Turn on MDM** button or the **My device** page presents you with an error, please contact your IT administrator.

> ## Migrate automatically enrolled (ADE) hosts

Reviewer: ```suggestion
## Migrate automatically enrolled (DEP) hosts
```

I think we use "DEP" in other spots in the product and docs. I could be wrong but worth double checking. IT admins are still used to DEP and we use it in internal/external convos.

Author: Ah gotcha! I assumed we were on "ADE" now, but lemme double check other docs we have.

Author: Yeah, looks like we use "DEP" a lot more than we use "ADE" in our docs. I will switch back to "DEP" for clarity

Reviewer 2: It's been almost a decade since Apple renamed this. We can use ADE.

Author: Happy to go either way! just lmk

Reviewer 2: Let's match Apple's language and use ADE.

Reviewer: Hey! I think let's prioritize consistency here and go w/ "DEP" for now.

At some point, we decided to use "DEP" b/c that's what we used in internal/external conversations. I think "DEP" is all over the product and docs.

It might be time to change to "ADE" but I think let's address that language change as a whole (everywhere) in a separate pass.

@jahzielv can you please help track a feature request for that so we don't lose it?

Reviewer 2: Excluding release notes, we use DEP in 3 other articles and ADE in 5 articles. Let's match Apple's language.

Reviewer: ADE sounds good to me 👍 

@spokanemac can you please help track a feature request to change it elsewhere? I think the important thing is we don't forget to make it consistent.

## Requirements

- A [deployed Fleet instance](https://fleetdm.com/docs/deploy/deploy-fleet)
- Fleet is connected to Apple Push Notification service (APNs) and Apple Business Manager (ABM). [See macOS MDM setup](https://fleetdm.com/guides/macos-mdm-setup)

## Migrate hosts

To migrate hosts, we will do the following steps:

1. Enroll hosts to Fleet
2. Assign hosts in Apple Business Manager (ABM) to Fleet
3. Choose migration workflow and migrate hosts

### Step 1: enroll hosts to Fleet

> 1. First your hosts to Fleet by installing Fleet's agent (fleetd). Learn how [here](https://fleetdm.com/guides/enroll-hosts).

Author: Missing "enroll"?
```suggestion
1. First enroll your hosts to Fleet by installing Fleet's agent (fleetd). Learn how [here](https://fleetdm.com/guides/enroll-hosts).
```

2. Ensure your end users have access to an admin account on their Mac. End users won't be able to migrate on their own if they have a standard account.

### Step 2: assign hosts in Apple Business Manager (ABM) to Fleet

> 1. In ABM, unassign the your hosts from your current MDM solution: select **Devices** and then select **All Devices**. Then, select **Edit** next to **Edit MDM Server**, select **Unassign from the current MDM**, and select **Continue**.

Author: ```suggestion
1. In ABM, unassign your hosts from your current MDM solution: select **Devices** and then select **All Devices**. Then, select **Edit** next to **Edit MDM Server**, select **Unassign from the current MDM**, and select **Continue**.
```

- If the host is not assigned to Fleet in ABM (manual enrollment), the end user will be given the option to download the MDM enrollment profile on their **My device page**.
<img width="1600" alt="Fleet icon in menu bar" src="https://raw.githubusercontent.com/fleetdm/fleet/main/website/assets/images/articles/fleet-desktop-says-hello-world-cover-1600x900@2x.jpg">
<img width="1400" alt="My device page - turn on MDM" src="https://user-images.githubusercontent.com/5359586/229950406-98343bf7-9653-4117-a8f5-c03359ba0d86.png">

Configuration:

- To kick off the default workflow, in your current MDM solution, unenroll the hosts to be migrated. macOS does not allow a host to be connected to multiple MDM solutions at once.

#### End user workflow

> Available in Fleet Premium

End user experience:

- To watch an animation of the end user experience during the migration workflow, in the Fleet UI, head to **Settings > Integrations > Mobile device management (MDM)**, and scroll down to the **End user migration workflow** section.

Configuration:

- In Fleet, you can configure the end user workflow using the Fleet UI, Fleet API, or Fleet's GitOps workflow.

> - After your configured the end user workflow, instruct your end users to select the Fleet icon in their menu bar, select **Migrate to Fleet** and follow the on-screen instructions to migrate to Fleet.

Author: ```suggestion
- After you configure the end user workflow, instruct your end users to select the Fleet icon in their menu bar, select **Migrate to Fleet** and follow the on-screen instructions to migrate to Fleet.
```

### Step 3: choose migration workflow and migrate hosts

There are two migration workflows in Fleet: default and end user.

The default migration workflow requires that the IT admin unenrolls hosts from the old MDM solution before the end user can complete migration. This will result in a gap in MDM coverage until the end user completes migration.

The end user migration workflow allows the end user to kick off migration by unenrolling from the old MDM solution on their own. Once the user is unenrolled, they're prompted to turn on MDM features in Fleet. This reduces the gap in MDM coverage.

#### Default workflow

End user experience:

Once a host is unenrolled from the old MDM solution, the end user will be given the option to download the MDM enrollment profile on their **My device page**.

![Fleet icon in menu bar](https://raw.githubusercontent.com/fleetdm/fleet/main/website/assets/images/articles/fleet-desktop-says-hello-world-cover-1600x900@2x.jpg)

<img width="1400" alt="My device page - turn on MDM" src="https://user-images.githubusercontent.com/5359586/229950406-98343bf7-9653-4117-a8f5-c03359ba0d86.png">

> If the host is in ABM, the end user will be prompted w/ Apple's **Remote Management** full-screen popup.

Reviewer: Hey @jahzielv is this how it works today? If a host is in ABM, assigned to Fleet in ABM, enrolled to Fleet, and unenerolled from the old MDM, Fleet pops up Apple's Remote Management dialog?

If so, I think we want to add a screenshot here so IT admins can see what that looks like.

Author: Hrm, good call out. I don't think the Remote Management dialog is popped up unless the host goes through what we're calling the "end-user workflow". The dialog only comes up if `profiles renew -type enrollment` is run (which we do via fleetd in the end-user workflow) I think we should remove this sentence.

Author: @noahtalerman nvm!! I just went through this flow and found that I was wrong.

The "Remote Management" dialog does in fact pop up after refetching the host. I forgot that Fleet checks (independently of the migration flow) if a host needs to run `profiles renew` and prompts fleetd to do so if needed.

I can add a screenshot.

Reviewer: Thank!

##### End user experience

> 1. The end user will receive a "Device Enrollment: &lt;organization&gt; can automatically configure your Mac." system notification within the macOS Notifications Center.

Reviewer: ```suggestion
1. The end user will receive a "Device Enrollment: &lt;organization&gt; can automatically configure your Mac." system notification within the macOS Notifications Center.
```

Is this still the case? I though the end user sees a full screen dialog? (not a system notification in Notifications Center)
